### PR TITLE
NO-DED updating LiveChat for Android document and correcting typos

### PR DESCRIPTION
**Pull request by kamilatokarska**
# Deploy preview

Scroll down to the checks section for the link to the deploy preview of your branch.

# Links

Link your Jira ticket.

- [Jira]()

# Description

PR including updates to the LiveChat for Android document. Additionally, typo mistake in the different document has been corrected (in all versions)

# Review and release

Apply changes and resolve conflicts. Once the described functionality lands on prod, let us know on #platform-docs-releases that your PR is ready for release. We will **merge** and **publish** the changes.

### CODE REVIEW

File: src/pages/getting-started/installing-livechat/android-widget/index.mdx [hunk 1]
```diff
@@ -319,6 +324,59 @@ android: configChanges = "orientation|screenSize";
 
 The chat widget will handle the orientation change by itself.
 
+# Localisation
```
Comment (kamilatokarska) [inline src/pages/getting-started/installing-livechat/android-widget/index.mdx:327]: This whole section was not in our docs. However, it looks like it was added to Szymon's README before. In the current PRs from him, there is `Migrating to versions...` section added. That is why I decided to add the whole `localisation` section to our docs.

File: src/pages/getting-started/installing-livechat/android-widget/index.mdx [hunk 2]
```diff
@@ -319,6 +324,59 @@ android: configChanges = "orientation|screenSize";
...
+
+### Migrating to version >= 2.2.0
+
+- ChatWindowView is now interface that can be casted to View
+- `setUpWindow(configuration);` is replaced by `setConfiguration(configuration);`
+- `setUpListener(listener)` is replaced by `setEventsListener(listener)`
+- `ChatWindowView.clearSession(Context)` is moved to `ChatWindowUtils.clearSession(Context)`
+- `ChatWindowView.createAndAttachChatWindowInstance(Activity)` is moved to `ChatWindowUtils.createAndAttachChatWindowInstance(getActivity())``
+
+# SnapCall integration
```
Comment (kamilatokarska) [inline src/pages/getting-started/installing-livechat/android-widget/index.mdx:345]: This whole section was added in one of the PRs

File: src/pages/getting-started/installing-livechat/android-widget/index.mdx [hunk 3]
```diff
@@ -319,6 +324,59 @@ android: configChanges = "orientation|screenSize";
...
+```java
+@Override
+public void onRequestAudioPermissions(String[] permissions, int requestCode) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+        this.requestPermissions(permissions, requestCode);
+    }
+}
+```
+
+4. Override your activity `void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults)` to pass result to `ChatWindowView`
```
Comment (kamilatokarska) [inline src/pages/getting-started/installing-livechat/android-widget/index.mdx:370]: I am not sure if it is possible to write it differently, but when you look at this sentence in the preview, it is too long, and it 'enters' the area of a sidebar 🤔

File: src/pages/management/configuration-api/index.mdx [hunk 4]
```diff
@@ -2495,7 +2495,7 @@ For `bot` webhooks, you need to [create a bot](#create-bot), [register webhooks]
...
 | `url`                                           | yes      | `string`   | Destination URL for the webhook                                                                                                                                                                                                                   |
-| `additional_data`                               | no       | `[]string` | Additional data arriving with the webhook. Value: `chat properties`, `chat_presence_user_ids` (all actions except for `agent_status_changed`, `agent_deleted`, `incoming_customer`)                                                               |
+| `additional_data`                               | no       | `[]string` | Additional data arriving with the webhook. Value: `chat_properties`, `chat_presence_user_ids` (all actions except for `agent_status_changed`, `agent_deleted`, `incoming_customer`)                                                               |
```
Comment (kamilatokarska) [inline src/pages/management/configuration-api/index.mdx:2498]: here and below, I just added one case from `platform-docs` as it is a tiny typo in all the versions but may cause problems for the users.

File: src/pages/getting-started/installing-livechat/android-widget/index.mdx [hunk 5]
```diff
@@ -319,6 +324,59 @@ android: configChanges = "orientation|screenSize";
 
 The chat widget will handle the orientation change by itself.
 
+# Localisation
+
+You can change or localize error messages, by defining your own string resources with following id's:
```
Comment (kamilatokarska) [inline src/pages/getting-started/installing-livechat/android-widget/index.mdx:329]:
```suggestion
You can change or localize error messages by defining your own string resources with the following id's:
```

File: src/pages/getting-started/installing-livechat/android-widget/index.mdx [hunk 6]
```diff
@@ -221,7 +224,9 @@ public boolean onError(ChatWindowErrorType errorType, int errorCode, String erro
...
+After your user signs out of the app, you might want to clear the chat session. You can do that by invoking a static method on `ChatWindowUtils.clearSession(Context)` from anywhere in the app. In case your `ChatWindowView` is attached in course of the log out flow, you also going to need to reload it by calling `chatWindow.reload(false)` after the `clearSession` code. See <a href="https://github.com/livechat/chat-window-android/blob/master/app/src/main/java/com/livechatinc/livechatwidgetexample/FullScreenWindowActivityExample.java" target="_blank" rel="noopener noreferrer">FullScreenWindowActivityExample.java</a>.
```
Comment (ale-kacperczyk) [inline src/pages/getting-started/installing-livechat/android-widget/index.mdx:227]: This whole section needs proofreading

File: src/pages/getting-started/installing-livechat/android-widget/index.mdx (hunk 3 above)
Comment (ale-kacperczyk) [reply src/pages/getting-started/installing-livechat/android-widget/index.mdx:370]: This is a bug :D Please create a Jira ticket (type bug) for this kind of issue. In this PR, try to break this line (enter + shift)

File: src/pages/getting-started/installing-livechat/android-widget/index.mdx (hunk 5 above)
Comment (ale-kacperczyk) [reply src/pages/getting-started/installing-livechat/android-widget/index.mdx:329]: I'd say "ids"

Comment (kamilatokarska) [reply src/pages/getting-started/installing-livechat/android-widget/index.mdx:329]: oooo, you are right!

File: src/pages/getting-started/installing-livechat/android-widget/index.mdx (hunk 6 above)
Comment (kamilatokarska) [reply src/pages/getting-started/installing-livechat/android-widget/index.mdx:227]: You got me - I admit that it was just `(false)` added here in one of the methods and a sentence at the end, and I assumed that the rest must be okay if it was here before. Will do it :)

File: src/pages/getting-started/installing-livechat/android-widget/index.mdx [hunk 7]
```diff
@@ -221,7 +224,9 @@ public boolean onError(ChatWindowErrorType errorType, int errorCode, String erro
...
+After your user signs out of the app, you might want to clear the chat session. You can do that by invoking a static method on `ChatWindowUtils.clearSession(Context)` from anywhere in the app. In case your `ChatWindowView` is attached during the logout flow, you will also need to reload it by calling `chatWindow.reload(false)` after the `clearSession` code. See <a href="https://github.com/livechat/chat-window-android/blob/master/app/src/main/java/com/livechatinc/livechatwidgetexample/FullScreenWindowActivityExample.java" target="_blank" rel="noopener noreferrer">FullScreenWindowActivityExample.java</a>.
+
+If your `ChatWindow` isn't recreated when `ChatWindowConfiguration` changes (i.e. VisitorName), you might want to fully reload chat window by invoking `chatWindow.reload(true)`.
```
Comment (ale-kacperczyk) [inline src/pages/getting-started/installing-livechat/android-widget/index.mdx:229]:
```suggestion
If your `ChatWindow` isn't recreated when `ChatWindowConfiguration` changes (i.e. VisitorName), you might want to fully reload the chat window by invoking `chatWindow.reload(true)`.
```

File: src/pages/getting-started/installing-livechat/android-widget/index.mdx [hunk 8]
```diff
@@ -319,6 +324,60 @@ android: configChanges = "orientation|screenSize";
...
+
+- ChatWindowView is now interface that can be casted to View
+- `setUpWindow(configuration);` is replaced by `setConfiguration(configuration);`
+- `setUpListener(listener)` is replaced by `setEventsListener(listener)`
+- `ChatWindowView.clearSession(Context)` is moved to `ChatWindowUtils.clearSession(Context)`
+- `ChatWindowView.createAndAttachChatWindowInstance(Activity)` is moved to `ChatWindowUtils.createAndAttachChatWindowInstance(getActivity())``
+
+# SnapCall integration
+
+SnapCall integration requires AUDIO and VIDEO permissions. In order to allow your users to use SnapCall integration you need to:
```
Comment (ale-kacperczyk) [inline src/pages/getting-started/installing-livechat/android-widget/index.mdx:347]:
```suggestion
The SnapCall integration requires AUDIO and VIDEO permissions. In order to allow your users to use the SnapCall integration, you need to:
```

File: src/pages/extending-agent-app/agent-app-sdk/index.mdx [hunk 9]
```diff
@@ -166,15 +166,15 @@ widget.on("customer_profile", profile => {
...
+| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `id`              | Unique ID of a visitor                                                                                                                                                                |
+| `name`            | Visitor name (if provided)                                                                                                                                                            |
+| `email`           | Visitor email (if provided)                                                                                                                                                           |
+| `chat`            | Object with three properties: `chat_id` (unique chat id), `groupID` (unique group id), and `id` (unique thread id); this object may be empty when a visitor is not currently chatting |
```
Comment (kamilatokarska) [inline src/pages/extending-agent-app/agent-app-sdk/index.mdx:174]: One more case from #platform-docs was added. Solutions agreed with Wojtek Zbrożek

**Review by ale-kacperczyk — APPROVED** [review]

**Pull request merged**